libraries/abstractions/common_io/test/test_iot_spi.c

```diff
@@ -200,25 +208,29 @@ TEST( TEST_IOT_SPI, AFQP_IotSPI_ReadSync )
     lRetVal = iot_spi_ioctl( xSPIHandle, eSPIGetMasterConfig, &xOrigConfig);
     TEST_ASSERT_EQUAL( IOT_SPI_SUCCESS, lRetVal );
 
-    /* configure bus */
-    xTestConfig.ulFreq = SPI_TEST_FREQ;
-    xTestConfig.eMode = xtestDefaultConfigMode;
-    xTestConfig.eSetBitOrder = xtestDefaultconfigBitOrder;
-    xTestConfig.ucDummyValue = 0;
-    lRetVal = iot_spi_ioctl( xSPIHandle, eSPISetMasterConfig, &xTestConfig);
-    TEST_ASSERT_EQUAL( IOT_SPI_SUCCESS, lRetVal );
+    if( TEST_PROTECT() )
+    {
+        /* configure bus */
+        xTestConfig.ulFreq = ultestIotSPIFrequency;
+        xTestConfig.eMode = xtestIotSPIDefaultConfigMode;
+        xTestConfig.eSetBitOrder = xtestIotSPIDefaultconfigBitOrder;
+        xTestConfig.ucDummyValue = ultestIotSPIDummyValue;
 
-    lRetVal = iot_spi_read_sync( xSPIHandle, ucRxBuf, 4);
-    TEST_ASSERT_EQUAL( IOT_SPI_SUCCESS, lRetVal );
+        lRetVal = iot_spi_ioctl( xSPIHandle, eSPISetMasterConfig, &xTestConfig);
+        TEST_ASSERT_EQUAL( IOT_SPI_SUCCESS, lRetVal );
 
-    //lRetVal = iot_spi_ioctl( xSPIHandle, eSPIGetRxNoOfbytes, &xBytesRx);
-    //TEST_ASSERT_EQUAL( IOT_SPI_SUCCESS, lRetVal );
-    //TEST_ASSERT_EQUAL( xBytesRx, 4 );
+        lRetVal = iot_spi_read_sync( xSPIHandle, ucRxBuf, 4);
+        TEST_ASSERT_EQUAL( IOT_SPI_SUCCESS, lRetVal );
+
+        lRetVal = iot_spi_ioctl( xSPIHandle, eSPIGetRxNoOfbytes, &xBytesRx);
```

Comment:
for sync APIs, RxNoOfBytes may not return the bytes. Sync API shall return success only if all the bytes are available.